src/preferences/colorpaletteeditormodel.cpp

```diff
@@ -53,25 +69,37 @@ bool ColorPaletteEditorModel::dropMimeData(const QMimeData* data, Qt::DropAction
 bool ColorPaletteEditorModel::setData(const QModelIndex& modelIndex, const QVariant& value, int role) {
     setDirty(true);
     if (modelIndex.isValid() && modelIndex.column() == 1) {
-        bool ok;
-        int hotcueIndex = value.toInt(&ok);
+        const bool initialAttemptSuccessful = QStandardItemModel::setData(modelIndex, value, role);
 
-        // Make sure that the value is valid
-        if (!ok || hotcueIndex <= 0 || hotcueIndex > rowCount()) {
-            return QStandardItemModel::setData(modelIndex, QVariant(), role);
+        const auto* hotcueIndexListItem = toHotcueIndexListItem(itemFromIndex(modelIndex));
```

Comment:
For pointers, we use the `p` prefix:

```suggestion
        const auto* pHotcueIndexListItem = toHotcueIndexListItem(itemFromIndex(modelIndex));
```